controllers/component_controller.go

```diff
@@ -189,8 +189,13 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 	} else {
 		if hasApplication.Status.Devfile != "" && (forceGenerateGitopsResource || len(component.Status.Conditions) > 0 && component.Status.Conditions[len(component.Status.Conditions)-1].Status == metav1.ConditionTrue && containsString(component.GetFinalizers(), compFinalizerName)) {
-			// only attempt to finalize and update the gitops repo if an Application is present & the previous Component status is good
+			// only attempt to finalize and update the gitops repo if an Application is present & not under deletion & the previous Component status is good
 			// A finalizer is present for the Component CR, so make sure we do the necessary cleanup steps
+			if !hasApplication.ObjectMeta.DeletionTimestamp.IsZero() {
+				log.Info("application %v is under deletion. Skipping deletion for component %v", hasApplication.Name, component.Name)
+				return ctrl.Result{}, nil
```

Comment:
Just had a thought.

If we return here, we never remove the finalizer off the component resource farther below:

https://github.com/redhat-appstudio/application-service/blob/52908910fc379bf5731a952c2f2093b886f93eb7/controllers/component_controller.go#L217

If that never happens, the Component CR can't be deleted, as the finalizer blocks deletion. We still need to remove the finalizer in this case.

Reply:
@johnmcollier I've updated the code in order to ensure that the finalizer is removed & we ignore the deletion for this case